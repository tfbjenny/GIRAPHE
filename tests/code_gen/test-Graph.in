node a = node("A");
node b = node("B");
node c = node("C");
node d = node("D");
node e = node("E");
node f = node("F");

node m = node("M");
node h = node("H");

node x = node("X");
node y = node("Y");
node z = node("Z");


print(" A -> E + A -> B -> D + A -> C ->F ");

graph g = a -> 1$e + a -> 2$b -> 3$d + a -> 4$c -> 5$f;
print(g);

print("Get Graph Size:");
print(g.size());

print("Add Nodes to Graph");
g.addNode(m);
print(g.size());
print(g);

print("Judge whether Nodes exist");
print(g.hasNode(x));
print(g.hasNode(a));

print("Add Edge to Graph, Support new Nodes");
g.addEdge(e, f, 5);
g.addEdge(a, z, 10);
print(g.size());
list<node> nodes = g.getAllNodes();
print(nodes);
print(g);

print("Judge whether Edges exist between two Nodes");
print(g.hasEdge(c,d));
print(g.hasEdge(a,b));



