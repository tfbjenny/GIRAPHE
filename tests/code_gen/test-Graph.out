 A -> E + A -> B -> D + A -> C ->F 
--------------------------------------
node A
node E
node B
node D
node C
node F
edgeA ->E: 1
edgeB ->D: 3
edgeA ->B: 2
edgeC ->F: 5
edgeA ->C: 4
--------------------------------------
Get Graph Size:
6
Add Nodes to Graph
7
--------------------------------------
node A
node E
node B
node D
node C
node F
node M
edgeA ->E: 1
edgeB ->D: 3
edgeA ->B: 2
edgeC ->F: 5
edgeA ->C: 4
--------------------------------------
Judge whether Nodes exist
false
true
Add Edge to Graph, Support new Nodes
8
[node A
node E
node B
node D
node C
node F
node M
node Z
]
--------------------------------------
node A
node E
node B
node D
node C
node F
node M
node Z
edgeA ->E: 1
edgeB ->D: 3
edgeA ->B: 2
edgeC ->F: 5
edgeA ->C: 4
edgeE ->F: 5
edgeA ->Z: 10
--------------------------------------
Judge whether Edges exist between two Nodes
false
true
